ministryofjustice/correspondence_tool_staff
# CT-2873 Changes to new data request form

## Description
This PR makes significant changes to the new Data Requests creation flow

* Removes the support for creating multiple data requests in a single page
* Adds pre-determined options for request type (changing the field name from `data`)
* Requires a mandatory, supplemental "notes" field if the request_type is `other`
* Adds optional "from" and "to" dates

This saves the data in the database on success, display and editing are added in other PRs

## Self-review checklist


### Screenshots

![image](https://user-images.githubusercontent.com/1161161/90442003-d1bf8e00-e0d1-11ea-8af4-0aaf0f18cf51.png)


### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CT-2873

### Deployment
This PR has database migrations that must be run using `rake db:migrate` to add and update columns on the DataRequest table

### Manual testing instructions
The database structure is not backwards compatible with master after running the migrations. Down migrations are supported, though, so if you need to run it and then switch back to master you'll need to do `rake db:rollback` for each database migration here; there are three. 

After running the database migrations, you should see the new screen when you click "record data request" on an offender SAR case. After saving the request, you should see the location and request type listed in the data requests table. You can check other values in the database.

Request type note should be mandatory if and only if "other" is chosen. 

The date fields are optional individually and together. The only validation is that the `from` date can't be after the `to` date
